tests/upgrades/test_repository.py

```diff
@@ -70,13 +62,15 @@ def test_pre_repository_scenario_upstream_authorization(self, target_sat, save_t
         rake_repo = f'repo = Katello::Repository.find_by_id({custom_repo})'
         rake_username = f'; repo.root.upstream_username = "{UPSTREAM_USERNAME}"'
         rake_repo_save = '; repo.save!(validate: false)'
-        result = run(f"echo '{rake_repo}{rake_username}{rake_repo_save}'|foreman-rake console")
-        assert 'true' in result
+        result = target_sat.execute(
+            f"echo '{rake_repo}{rake_username}{rake_repo_save}'|foreman-rake console"
+        )
+        assert 'true' in result.stdout
```

Comment:
I think it would be nice to have some helper for running such rake console commands, non-blocking

Reply:
okay that would go system info helper